src/hipscat/io/validation.py

```diff
@@ -16,13 +33,129 @@ def is_valid_catalog(pointer: FilePointer, storage_options: Union[Dict[Any, Any]
         True if both the catalog_info and partition_info files are
         valid, False otherwise
     """
-    return is_catalog_info_valid(pointer, storage_options=storage_options) and (
-        is_partition_info_valid(pointer, storage_options=storage_options)
-        or is_metadata_valid(pointer, storage_options=storage_options)
+    if not strict:
+        return is_catalog_info_valid(pointer, storage_options=storage_options) and (
+            is_partition_info_valid(pointer, storage_options=storage_options)
+            or is_metadata_valid(pointer, storage_options=storage_options)
+        )
+
+    if verbose:
+        print(f"Validating catalog at path {pointer} ... ")
+
+    is_valid = True
+
+    def handle_error(msg):
+        """inline-method to handle repeated logic of raising error or warning and
+        continuing."""
+        nonlocal fail_fast
+        nonlocal is_valid
+        nonlocal verbose
+        if fail_fast:
+            raise ValueError(msg)
+        if verbose:
+            print(msg)
+        else:
+            warnings.warn(msg)
+        is_valid = False
+
+    if not is_catalog_info_valid(pointer, storage_options=storage_options):
+        handle_error("catalog_info.json file is invalid.")
+
+    if not is_partition_info_valid(pointer, storage_options=storage_options):
+        handle_error("partition_info.csv file is invalid.")
+
+    if not is_metadata_valid(pointer, storage_options=storage_options):
+        handle_error("_metadata file is invalid.")
+
+    if not is_common_metadata_valid(pointer, storage_options=storage_options):
+        handle_error("_common_metadata file is invalid.")
```

Comment:
Currently these methods only check if the files exist. Can we change the messages to make it clear?